jpyinterpreter/src/main/java/ai/timefold/jpyinterpreter/types/numeric/PythonDecimal.java

```diff
@@ -0,0 +1,807 @@
+package ai.timefold.jpyinterpreter.types.numeric;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.function.BiPredicate;
+import java.util.stream.Collectors;
+
+import ai.timefold.jpyinterpreter.PythonClassTranslator;
+import ai.timefold.jpyinterpreter.PythonLikeObject;
+import ai.timefold.jpyinterpreter.PythonOverloadImplementor;
+import ai.timefold.jpyinterpreter.types.AbstractPythonLikeObject;
+import ai.timefold.jpyinterpreter.types.BuiltinTypes;
+import ai.timefold.jpyinterpreter.types.PythonLikeType;
+import ai.timefold.jpyinterpreter.types.PythonNone;
+import ai.timefold.jpyinterpreter.types.PythonString;
+import ai.timefold.jpyinterpreter.types.collections.PythonLikeTuple;
+import ai.timefold.jpyinterpreter.types.errors.TypeError;
+import ai.timefold.jpyinterpreter.types.errors.ValueError;
+import ai.timefold.solver.core.impl.domain.solution.cloner.PlanningImmutable;
+
+public class PythonDecimal extends AbstractPythonLikeObject implements PythonNumber, PlanningImmutable {
+    public final BigDecimal value;
+    private static final ThreadLocal<MathContext> threadMathContext =
+            ThreadLocal.withInitial(() -> new MathContext(28, RoundingMode.HALF_EVEN));
```

Comment:
This is never modified. Why is it a thread-local instead of a constant?

Reply:
To make it easier when `setcontext` methods are added (which are thread-local)

Reply:
I'd rather we avoid this. If we need a field on the object, let's create one.

Reply:
Different threads can have different contexts, and it not bound to instances. This allows code like this in Python:

```python
with localcontext() as ctx:
    ctx.prec = 42   # Perform a high precision calculation
    s = calculate_something()
```
(https://docs.python.org/3/library/decimal.html#decimal.localcontext)